tsl/src/partialize_agg.c

```diff
@@ -0,0 +1,115 @@
+/*
+ * This file and its contents are licensed under the Timescale License.
+ * Please see the included NOTICE for copyright information and
+ * LICENSE-TIMESCALE for a copy of the license.
+ */
+
+#include <postgres.h>
+#include <catalog/pg_operator.h>
+#include <miscadmin.h>
+#include <nodes/bitmapset.h>
+#include <nodes/makefuncs.h>
+#include <nodes/nodeFuncs.h>
+#include <optimizer/cost.h>
+#include <optimizer/optimizer.h>
+#include <optimizer/pathnode.h>
+#include <optimizer/paths.h>
+#include <parser/parsetree.h>
+#include <utils/builtins.h>
+#include <utils/lsyscache.h>
+#include <utils/typcache.h>
+#include <utils/fmgroids.h>
+
+#include <planner.h>
+
+#include "nodes/decompress_chunk/decompress_chunk.h"
+#include "partialize_agg.h"
+#include "utils.h"
+#include "debug_assert.h"
+
+#define is_restricted_path(path) list_length(path->parent->baserestrictinfo)
```

Comment:
Thanks. Added the join clauses to the test. Since we test two paths, I will keep the macro for now. But when the second path test is removed in https://github.com/timescale/timescaledb/pull/6218, I will also remove the macro.